src/mapping/parser/mod.rs

```diff
@@ -0,0 +1,815 @@
+extern crate pest;
+
+use crate::{
+    event::Value,
+    mapping::{
+        query::{
+            self,
+            arithmetic::Arithmetic,
+            arithmetic::Operator,
+            functions::{ToBooleanFn, ToFloatFn, ToIntegerFn, ToStringFn, ToTimestampFn},
+            path::Path as QueryPath,
+            Literal,
+        },
+        Assignment, Deletion, Function, IfStatement, Mapping, Noop, OnlyFields, Result,
+    },
+};
+use pest::{
+    error::ErrorVariant,
+    iterators::{Pair, Pairs},
+    Parser,
+};
+
+#[derive(Parser)]
+#[grammar = "./mapping/parser/grammar.pest"]
+struct MappingParser;
+
+fn target_path_from_pair(pair: Pair<Rule>) -> String {
+    let mut segments = Vec::new();
+    for segment in pair.into_inner() {
+        match segment.as_rule() {
+            Rule::path_segment => segments.push(segment.as_str().to_string()),
+            Rule::quoted_path_segment => {
+                segments.push(quoted_path_from_pair(segment).replace(".", "\\."))
+            }
+            _ => unreachable!(),
```

Comment:
Please give this a parameter describing what a user should do if they see this unreachable. (Report a bug?)